lib/wireprotocol/2_6_support.js

```diff
@@ -275,94 +198,54 @@ var setupClassicFind = function(bson, ns, cmd, cursorState, topology, options) {
     ignoreUndefined: ignoreUndefined
   });
 
-  // Set query flags
-  query.slaveOk = readPreference.slaveOk();
-
-  // Set up the option bits for wire protocol
-  if (typeof cmd.tailable === 'boolean') {
-    query.tailable = cmd.tailable;
-  }
-
-  if (typeof cmd.oplogReplay === 'boolean') {
-    query.oplogReplay = cmd.oplogReplay;
-  }
-
-  if (typeof cmd.noCursorTimeout === 'boolean') {
-    query.noCursorTimeout = cmd.noCursorTimeout;
-  }
-
-  if (typeof cmd.awaitData === 'boolean') {
-    query.awaitData = cmd.awaitData;
-  }
-
-  if (typeof cmd.partial === 'boolean') {
-    query.partial = cmd.partial;
-  }
+  if (typeof cmd.tailable === 'boolean') query.tailable = cmd.tailable;
+  if (typeof cmd.oplogReplay === 'boolean') query.oplogReplay = cmd.oplogReplay;
+  if (typeof cmd.noCursorTimeout === 'boolean') query.noCursorTimeout = cmd.noCursorTimeout;
+  if (typeof cmd.awaitData === 'boolean') query.awaitData = cmd.awaitData;
+  if (typeof cmd.partial === 'boolean') query.partial = cmd.partial;
 
-  // Return the query
+  query.slaveOk = readPreference.slaveOk();
   return query;
-};
+}
 
-//
-// Set up a command cursor
-var setupCommand = function(bson, ns, cmd, cursorState, topology, options) {
-  // Set empty options object
+function setupCommand(bson, ns, cmd, cursorState, topology, options) {
   options = options || {};
-  // Get the readPreference
-  var readPreference = getReadPreference(cmd, options);
+  const readPreference = getReadPreference(cmd, options);
+  const parts = ns.split(/\./);
```

Comment:
probably exists from before, but why is this a regex and not just `'.'`?

Reply:
believe me this is all changing by the 5th PR on top of this one 😄 I refactored this all into `databaseNamespace` and `collectionNamespace`, all of the splitting and joining is hidden away and done without regular expressions

Reply:
nice :)